src/team_lib/mhba/xccl_mhba_lib.c

```diff
@@ -340,14 +352,17 @@ xccl_mhba_team_create_post(xccl_tl_context_t *context,
     xccl_sbgp_t *node, *net;
     struct ibv_qp_init_attr qp_init_attr;
     struct ibv_port_attr port_attr;
-    int shmid, i;
+    int shmid, i,mhba_data_size;
     size_t storage_size, local_data_size;
     uint32_t *local_data, *global_data;
+    mhba_team->context = ctx;
 
     XCCL_TEAM_SUPER_INIT(mhba_team->super, context, params, base_team);
     node = xccl_team_topo_get_sbgp(base_team->topo, XCCL_SBGP_NODE);
     mhba_team->node.sbgp = node;
-    storage_size = (MHBA_CTRL_SIZE+MHBA_DATA_SIZE) * node->group_size;
+    mhba_team->node.size_of_data_unit = sizeof(struct ibv_mr);
```

Comment:
why do you need "size_of_data_unit" as variable? can it be changed? 
keep 
#define MHBA_DATA_SIZE sizeof(struct ibv_mr)